Aleksandrsid 35519 bjs2 68746

AlexandrSid opened the pull request:
Допы в задаче на стратегию + тесты

AlexandrSid committed:
Task BJS2-68746
Task structure complete

AlexandrSid committed:
Task BJS2-68746
Basic logic complete

AlexandrSid committed:
Task BJS2-68746
Additional logic complete.
Small access modifiers fix.

AlexandrSid committed:
Task BJS2-68746
All methods tested

miked4king commented on src/main/java/school/faang/projectteammanagement/ProjectManager.java:
```diff
@@ -0,0 +1,85 @@
+
+    //добавляет сотрудника в систему. (без проверки уникальности)
+    public boolean addEmployee(Employee employee) {
+        this.allEmployees = new ArrayList<>(allEmployees);
```
а зачем создавать новый лист? 
давай в текущий добавим

AlexandrSid replied:
да, тут не место. Это чтобы в случае передачи fixed-size листа можно было потом в него добавлять сотрудников. Перенёс в тесты

miked4king commented on src/main/java/school/faang/projectteammanagement/ProjectManager.java:
```diff
@@ -0,0 +1,85 @@
+
+    public void assignTeamToProject(int projectId) {
+        getProjectById(projectId).setTeamMembers(
+                activeStrategy.assignTeam(getProjectById(projectId), allEmployees)
```
передавать в метод вызов другого метода не очень хорошая практика, код становится труднее понимать.
давай на такие случаи результат метода выносить в отдельную переменную и в дальнейшем передавать уже ее

miked4king commented on src/main/java/school/faang/projectteammanagement/ProjectManager.java:
```diff
@@ -0,0 +1,85 @@
+    public List<Project> findProjectsForEmployee(Employee employee) {
+        return companyProjects.stream()
+                .filter(p -> p.getRequiredSkills().stream().anyMatch(employee.getSkills()::contains))
+                .collect(Collectors.toList());
```
давай сделаем более информативное название параметра лямбды, чтобы другим разработчикам было легче понимать чем мы оперируем
в 17 java можно вызывать вместо collect(Collectors.toList()) просто toList()

miked4king commented on src/main/java/school/faang/projectteammanagement/ProjectManager.java:
```diff
@@ -0,0 +1,85 @@
+    public void removeEmployeeFromProject(int projectId, int employeeId) {
+        Project projectById = getProjectById(projectId);
+        projectById.setTeamMembers(
+                projectById.getTeamMembers().stream().filter(e -> e.getId() != employeeId).toList()
```
вынесем результат в переменную

miked4king commented on src/main/java/school/faang/projectteammanagement/ProjectManager.java:
```diff
@@ -0,0 +1,85 @@
+                .stream()
+                .filter(project -> (project.getProjectId() == id))
+                .findFirst()
+                .get();
```
вот тут не совсем корректно.
никогда не рекомендуется вызывать get на Optional без проверки, т.к можем упасть в случае если он будет пустой.

miked4king commented on src/main/java/school/faang/projectteammanagement/ProjectManager.java:
```diff
@@ -0,0 +1,85 @@
+                .collect(Collectors.toList());
+    }
+
+    //добавляет сотрудника в команду проекта, если у него есть все требуемые навыки.
```
не забудь убрать комментарии перед отправкой)

AlexandrSid committed:
Task BJS2-68746
changes based on PR review
comments removed
Optional driven logic added
code clean up

AlexandrSid commented:
https://faang-school.atlassian.net/browse/BJS2-68746
